test/05-device-state.spec.coffee

```diff
@@ -290,6 +290,13 @@ describe 'deviceState', ->
 			done()
 		, 5
 
+	it 'cancels current promise applying the target state', ->
+		@deviceState.applyInProgress = Promise.finally => @applyInProgress = null
```

Comment:
`Promise.finally` is actually undefined. I think you could instead do:

```
@deviceState.applyInProgress = (new Promise(->)).finally => @deviceState.applyInProgress = null
```

Reply:
Thank you. You're of course 100% correct.

Reply:
I changed it slighly to make it testable due to the asynchronous nature of promises. Feel free to comment on my approach. Not sure if this is the ideal way